bson/src/main/scala/org/mongodb/scala/bson/BsonValue.scala

```diff
@@ -0,0 +1,436 @@
+   *
+   * @param json the json string
+   * @return the BsonDocumebt
+   */
+  def apply(json: String): BsonDocument = JBsonDocument.parse(json)
+}
+
+/**
+ * Companion helper for a BsonDouble
+ * @since 1.0
+ */
+object BsonDouble {
+  /**
+   * Creates a `BsonDouble`
+   *
+   * @param value the BsonDouble value
+   * @return the BsonDouble
+   */
+  def apply(value: Double): BsonDouble = new BsonDouble(value)
+}
+
+/**
+ * Companion helper for a BsonInt32
+ * @since 1.0
+ */
+object BsonInt32 {
+  /**
+   * Creates a `BsonInt32`
+   *
+   * @param value the BsonInt32 value
+   * @return the BsonInt32
+   */
+  def apply(value: Int): BsonInt32 = new BsonInt32(value)
+}
+
+/**
+ * Companion helper for a BsonInt64
+ * @since 1.0
+ */
+object BsonInt64 {
+  /**
+   * Creates a `BsonInt64`
+   *
+   * @param value the BsonInt64 value
+   * @return the BsonInt64
+   */
+  def apply(value: Long): BsonInt64 = new BsonInt64(value)
+}
+
+/**
+ * Companion helper for a BsonJavaScript
+ * @since 1.0
+ */
+object BsonJavaScript {
+  /**
+   * Creates a `BsonJavaScript`
+   *
+   * @param value the javascript function
+   * @return the BsonJavaScript
+   */
+  def apply(value: String): BsonJavaScript = new BsonJavaScript(value)
+}
+
+/**
+ * Companion helper for a BsonJavaScriptWithScope
+ * @since 1.0
+ */
+object BsonJavaScriptWithScope {
+  import BsonMagnets._
+
+  /**
+   * Creates a `BsonJavaScript`
+   *
+   * @param value the javascript function
+   * @param scope the function scope
+   * @return the BsonJavaScript
+   */
+  def apply(value: String, scope: BsonDocument): BsonJavaScriptWithScope = new BsonJavaScriptWithScope(value, scope)
+
+  /**
+   * Creates a `BsonJavaScript`
+   *
+   * @param value the javascript function
+   * @param scope the function scope
+   * @return the BsonJavaScript
+   */
+  def apply(value: String, scope: CanBeBsonElement*): BsonJavaScriptWithScope = new BsonJavaScriptWithScope(value, BsonDocument(scope: _*))
+
+  /**
+   * Creates a `BsonJavaScript`
+   *
+   * @param value the javascript function
+   * @param scope the function scope
+   * @return the BsonJavaScript
+   */
+  def apply(value: String, scope: Traversable[(String, BsonValue)]): BsonJavaScriptWithScope = new BsonJavaScriptWithScope(value, BsonDocument(scope))
+}
+
+/**
+ * Companion helper for a BsonMaxKey
+ * @since 1.0
+ */
+object BsonMaxKey {
+  /**
+   * Creates a `BsonMaxKey`
+   * @return the BsonMaxKey
+   */
+  def apply(): BsonMaxKey = new BsonMaxKey()
+}
+
+/**
+ * Companion helper for a BsonMinKey
+ * @since 1.0
+ */
+object BsonMinKey {
+  /**
+   * Creates a `BsonMinKey`
+   * @return the BsonMinKey
+   */
+  def apply(): BsonMinKey = new BsonMinKey()
+}
+
+/**
+ * Companion helper for a BsonNull
+ * @since 1.0
+ */
+object BsonNull {
+  /**
+   * Creates a `BsonNull`
+   * @return the BsonNull
+   */
+  def apply(): BsonNull = new BsonNull()
+}
+
+/**
+ * Companion helper for a BsonNumber
+ * @since 1.0
+ */
+object BsonNumber {
```

Comment:
Not strictly necessary, but as companion objects are just convenience helpers it might prove convenient.